### PR TITLE
[4.0] J Installation: Install Languages. Make sure the namespace autoloader exists

### PR DESCRIPTION
**Pull request by roland-d**
### Summary of Changes
Trying to install a language during the installation routine, it requires the `JNamespacePsr4Map` class. However this class is never loaded in the installation routine.

### Testing Instructions
1. Start the Joomla installation routine
2. When Joomla is installed you get to the following page:
image
3. Click on the `Customize Installation`
4. Select the `French` language and scroll all the way down
5. Open the Developer Console (F12)
6. Open the Network tab in the Developer console
7. Click Next
8. You are now back on the page saying `Congratulations! Your Joomla site is ready.`
9. In the Network tab, click on the URL and you see this:
image
10. Notice the text `"message":"Class 'JNamespacePsr4Map' not found"}` and the `"error":true`
11. Apply the patch
12. Click on the `Customize Installation`
13. Select the `French` language and scroll all the way down
14. Click on Next
15. You are now back on the page saying `Congratulations! Your Joomla site is ready.`
16. Click on the `Customize Installation`
17. Notice that French is no longer in the list of available languages to install. The languages has now been installed.
18. You can also check the Developer console as it now looks like this:
image
19. Notice the `"error":false`
20. The issue has now been fixed :)

### Expected result
The language installs without error

### Actual result
The language does not install due to an error

### Documentation Changes Required
None

### CODE REVIEW

**Comment by infograf768** [comment]
This breaks install of a clean 4.0 here:
Screen Shot 2019-06-05 at 10 09 13

**Comment by infograf768** [comment]
@roland-d 
I solved the issue by moving
```
		// Ensure we load the namespace loader
		$this->createExtensionNamespaceMap();
```
to the `doExecute()` method.
i.e.
```
	/**
	 * Method to run the Web application routines.
	 *
	 * @return  void
	 *
	 * @since   3.1
	 */
	protected function doExecute()
	{
		// Ensure we load the namespace loader
		$this->createExtensionNamespaceMap();

		// Initialise the application.
		$this->initialiseApp();

		// Dispatch the application.
		$this->dispatch();
	}
```

as done in
`/build/stubGenerator.php`

**Comment by roland-d** [comment]
@infograf768 When do you get this? When I open the Joomla site I get this:
image

This is the URL `/installation/index.php`

**Comment by Quy** [comment]
I got the same error. 
Download your branch. 
Run `composer install` and `npm ci`. 
Proceed to install in browser:
`http://localhost/joomla-cms-feature-language-installation-error/installation/index.php`

**Comment by roland-d** [comment]
@Quy, I followed your steps and still get the normal installation screen.

The reason I put it in the `InstallationApplication` is because in the main CMS this code is also located in the `CMSApplication` and `ConsoleApplication`. So the `InstallationApplication` looks like the logical place to put it.

I would like to know what the actual error is. The message shown now is just a user message and not the actual error itself.

Later I will test it on another machine and see what happens there.

**Comment by infograf768** [comment]
It is OK to place it in InstallationApplication imho.
It is the placement of `$this->createExtensionNamespaceMap();` which has solved the issue here.
Can you test with my proposal?

**Comment by roland-d** [comment]
@infograf768 Putting the code in `doExecute()` also works for me. I just really wonder why it in the `__construct()` works for me.

Let's keep it in doExecute if it works for the majority :)

File: installation/src/Application/InstallationApplication.php [hunk 1]
```diff
@@ -37,6 +37,8 @@
  */
 final class InstallationApplication extends CMSApplication
 {
+	use \Joomla\CMS\Application\ExtensionNamespaceMapper;
```
Comment (Quy) [inline installation/src/Application/InstallationApplication.php:40]: Should this be with the other `use` statements?

Comment (roland-d) [reply installation/src/Application/InstallationApplication.php:40]: Not in this case because the `ExtensionNamespaceMapper` is a trait.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 58fff76e7421813d4c010e5f75d8fcbbc1508985<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25121">issues.joomla.org/tracker/joomla-cms/25121</a>.</sub>

**Comment by infograf768** [comment]
@Quy 
Can you please test again so that we get at last rid of this issue?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 22482d2f2bd3f460de5bc0f842c52d8ddfa98e0a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25121">issues.joomla.org/tracker/joomla-cms/25121</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25121">issues.joomla.org/tracker/joomla-cms/25121</a>.</sub>

**Comment by infograf768** [comment]
Thanks!

**Pull request merged**